### DISCUSSION TITLE
How do you feel about using Lombok for the getters and setters

### DISCUSSION THREAD

**Discussion by TCke83** [Smarteon/loxone-java#160] [Q&A] [answered]
I'm a frequent user of the lombok library and would like to introduce it as dependency into the library.

Do you think this is a good idea?

Let me know.

**Comment by jimirocks** [comment]
not very familiar with Lombok, but not sure whether it is good fit 
1. most of the objects are immutable, so the current situation with constructor based deserialization and getters doesn't bother much. 
2. not sure how well is Lombok compatible with Kotlin and Android
3. how much dependencies come with that - we try to keep the dependency list as short as possible

@xzavrel Could you please add your opinion on that?

**Comment by jimirocks** [comment, marked as answer]
Well, I have checked the thing more deeply and I would say we can try do go with Lombok. As it seems it's only build time dependency. 

So you can try to introduce it in some you future contributions
